### PR TITLE
Make use of the uvicorn reload feature

### LINKED ISSUE #1108: Remove `hupper` from debug server
Looks like uvicorn has a reload option that we can use instead of building our own reload mechanism: https://github.com/encode/uvicorn/blob/b5af1049e63c059dc750a450c807b9768f91e906/uvicorn/main.py#L410-L431

### PR DESCRIPTION
**Pull request by DoctorJohn**
## Description

This PR refactors the debug servers reload feature to use `uvicorns` reloader instead of `hupper`. See #1108 for more details.

Uvicorn requires us to pass the asgi application as an import string to enable the reload feature. We need a way to configure the asgi app before it's served by uvicorn with the strawberry schema supplied by the user via the command line.

Three options came into my mind on how to pass the user supplied schema to the asgi app:

1. Code generation (eww)
2. Letting the ASGI app parse the CLI arguments again (sys.argv)
3. Passing the schema import string using an environment variable (os.environ)

I tried the second approche first, it worked but felt too error prone. The third option is the one currently implemented in this PR. Works well, doesn't feel to hacky.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #1108

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release refactors the reload feature of the inbuilt debug server. The refactor
made the debug server more responsive and allowed us to remove `hupper` from the
dependencies.

**Comment by scratchmex** [comment]
There is a vulnerability with the bot. Maintainers, contact me and I will share details.

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Release type: patch
+
+This release refactors the reload feature of the inbuilt debug server. The refactor
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This release refactors the reload feature of the built-in debug server. The refactor
```

**Review by patrick91 — APPROVED** [review]
Using an environment variable looks ok to me! Much better than using hupper to be honest 😊

I'd wait for a double check from @jkimbo and @BryceBeagle though 😊

File: tests/cli/conftest.py [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
-    def watch_files(*args, **kwargs):
-        return True
+@pytest.fixture
+def debug_server_client(mocker):
+    schema_import_path = "tests.fixtures.sample_package.sample_module"
+    mocker.patch.object(sys, "argv", ["strawberry", "server", schema_import_path])
+
+    from strawberry.cli.debug_server import app
+
+    return TestClient(app)
```
Comment (BryceBeagle) [inline tests/cli/conftest.py:24]: Maybe I'm missing something but this test doesn't use the env var and instead uses a CLI argument?

**Review by BryceBeagle — COMMENTED** [review]
Is it possible to write a test that checks the auto-reload feature?

**Comment by DoctorJohn** [comment]
> Is it possible to write a test that checks the auto-reload feature?

Just spent 3 hours on it 🤯 There are a lot of moving parts involved: creating and updating a file containing the schema, running the actual debug server in a subprocess and sending actual requests to it.

I expected the test to end up being way too flaky, but it works just fine :)

**Comment by patrick91** [comment]
Thanks @DoctorJohn <3

**Pull request merged**